src/internal/m365/collection/drive/collections.go

```diff
@@ -714,6 +722,95 @@ func (c *Collections) getCollectionPath(
 	return collectionPath, nil
 }
 
+type driveEnumerationStats struct {
+	numPages      int
+	numAddedFiles int
+	numContainers int
+	numBytes      int64
+}
+
+func newPagerLimiter(opts control.Options) *pagerLimiter {
+	res := &pagerLimiter{
+		isPreview:            opts.ToggleFeatures.PreviewBackup,
+		maxContainers:        opts.ItemLimits.MaxContainers,
+		maxItemsPerContainer: opts.ItemLimits.MaxItemsPerContainer,
+		maxItems:             opts.ItemLimits.MaxItems,
+		maxBytes:             opts.ItemLimits.MaxBytes,
+		maxPages:             opts.ItemLimits.MaxPages,
+	}
+
+	if res.maxContainers == 0 {
+		res.maxContainers = defaultPreviewNumContainers
+	}
+
+	if res.maxItemsPerContainer == 0 {
+		res.maxItemsPerContainer = defaultPreviewNumItemsPerContainer
+	}
+
+	if res.maxItems == 0 {
+		res.maxItems = defaultPreviewNumItems
+	}
+
+	if res.maxBytes == 0 {
+		res.maxBytes = defaultPreviewNumBytes
+	}
+
+	if res.maxPages == 0 {
+		res.maxPages = defaultPreviewNumPages
+	}
+
+	return res
+}
+
+type pagerLimiter struct {
+	isPreview            bool
+	maxContainers        int
+	maxItemsPerContainer int
+	maxItems             int
+	maxBytes             int64
+	maxPages             int
+}
+
+func (l pagerLimiter) enabled() bool {
+	return l.isPreview
+}
+
+// sizeLimit returns the total number of bytes this backup should try to
+// contain.
+func (l pagerLimiter) sizeLimit() int64 {
+	return l.maxBytes
+}
```

Comment:
Either way works, take it or leave it.
```suggestion
// isTooLarge returns true if the item byte size is greater than the maximum allowed limit
func (l pagerLimiter) isTooLarge(item getSizer) bool {
	return ptr.Val(item.GetSize()) >= l.maxBytes
}
```

Reply:
fair.  Fwiw, the spirit of the suggestion is intended to ask "can we condense the check into this func", and not suggest changing the check parameters itself.